File: integration-test/java/spring-app/src/main/java/com/qualityminds/lazyval/integration/boundary/persistence/OrderDateToDateConverter.java
package com.qualityminds.lazyval.integration.boundary.persistence;

import com.qualityminds.lazyval.integration.domain.OrderDate;
import org.springframework.core.convert.converter.Converter;
import org.springframework.data.convert.WritingConverter;

import java.time.ZoneOffset;
import java.util.Date;

@WritingConverter
class OrderDateToDateConverter implements Converter<OrderDate, Date> {

    OrderDateToDateConverter() {
    }

    @Override
    public Date convert(OrderDate source) {
        return Date.from(source.value().atStartOfDay(ZoneOffset.UTC).toInstant());
    }
}

File: integration-test/kotlin/spring-app/src/main/kotlin/com/qualityminds/lazyval/integration/boundary/persistence/OrderDateToDateConverter.kt
package com.qualityminds.lazyval.integration.boundary.persistence

import com.qualityminds.lazyval.integration.domain.OrderDate
import org.springframework.core.convert.converter.Converter
import org.springframework.data.convert.WritingConverter
import java.time.ZoneOffset
import java.util.*

@WritingConverter
internal class OrderDateToDateConverter : Converter<OrderDate, Date> {
    override fun convert(source: OrderDate): Date =
        Date.from(source.value.atStartOfDay(ZoneOffset.UTC).toInstant())
}
